src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs

```diff
@@ -92,6 +93,16 @@ public static class AspireAzureCosmosDBExtensions
             });
         }
 
+        if (settings.IgnoreEmulatorCertificate && CosmosUtils.IsEmulatorConnectionString(settings.ConnectionString))
+        {
+            clientOptions.HttpClientFactory = () => new HttpClient(new HttpClientHandler()
+            {
+                ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
+            });
+            clientOptions.ConnectionMode = ConnectionMode.Gateway;
+            clientOptions.LimitToEndpoint = true;
```

Comment:
This line isn't in 

https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-develop-emulator?tabs=windows%2Ccsharp&pivots=api-nosql#connect-to-the-emulator-from-the-sdk

It just has the `ServerCertificateCustomValidationCallback` and `ConnectionMode` lines. Why is `LimitToEndpoint` being set to true here?

Reply:
I'm not 100% sure of the reason why but without `LimitToEndpoint` the trick doesn't work. I think it has something to do with the way that the Cosmos client talks to different regional gateways. But I'm not sure how/why that comes into play in the emulator scenario ... but it does seem to make a difference here.

Reply:
Should the docs be updated then? - cc @Pilchie @sourabh1007

Reply:
Looks like document is already updated and `LimitToEndpoint` is not required.

Reply:
@eerhardt docs was update just a day back or so. 
@mitchdenny might have referenced it before that change. 

Emulator will only have a single region; functional wise it will not impact but unnecessary and confuses.

Ability to ignore SSLCert through connection string is currently in PR stage and next release might ship it. Post that hopefully all above can be updated accordingly.

Reply:
@mitchdenny - can you verify that we don't need the line setting `LimitToEndpoint = true;`? I don't want us doing something that isn't in the docs.

Reply:
Whoever wrote those docs did not consult with the SDK team. The original version of that doc had it and that might have brought the confusion. We discovered the issue, we corrected the docs (https://github.com/MicrosoftDocs/azure-docs-pr/pull/263406). `LimitToEndpoint` has nothing to do with emulator or SSL certificates, it's a flag meant to **disable cross-region retries** on the SDK -> https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/troubleshoot-sdk-availability 
![image](https://github.com/dotnet/aspire/assets/1633401/3f0b26bc-773e-43a8-aa20-754a6b1acc10)